qiskit/circuit/library/standard_gates/global_phase.py

```diff
@@ -61,3 +64,12 @@ def __array__(self, dtype=complex):
         """Return a numpy.array for the global_phase gate."""
         theta = self.params[0]
         return numpy.array([[numpy.exp(1j * theta)]], dtype=dtype)
+
+    def __eq__(self, other):
+        if isinstance(other, GlobalPhaseGate):
+            try:
+                if math.isclose(self.params[0], other.params[0], rel_tol=0, abs_tol=1e-10):
+                    return True
+            except TypeError:
+                return self.params[0] == other.params[0]
```

Comment:
I see this is a reworking of the current scalar comparison within `Instruction.__eq__`.  Is it maybe worth centralising this into a separate function within `Instruction` (or just within the same file), so we can share the same scalar logic in all places that use it?

Reply:
It probably is worth it. I was into sunk cost fallacy around `RYGate` and just kept copying the logic through to all the gates.

Reply:
Updated in: https://github.com/Qiskit/qiskit/pull/11370/commits/d46cf05cc6011d16378e911c113c4f388177f9d0